src/hip_memory.cpp

```diff
@@ -481,13 +481,14 @@ hipError_t ihipHostMalloc(TlsData *tls, void** ptr, size_t sizeBytes, unsigned i
         hipDeviceSynchronize();
     }
 
+    if (sizeBytes == 0) {
+        // TODO - should size of 0 return err or be siliently ignored?
+        return hipSuccess;
+    }
```

Comment:
Here and all below: do the check on entering the function; wrap the return code with `ihipLogStatus`.
```c++
if (sizeBytes == 0) return ihipLogStatus(hipSuccess);
```